apis/helpers/namer_test.go

```diff
@@ -7,57 +7,74 @@ import (
+	shortBase := randSeq(validation.DNS1123SubdomainMaxLength - hashLen + 1)
 
-		tests := []struct {
-			base, suffix, expected string
-		}{
-			{
-				base:     shortName,
-				suffix:   "deploy",
-				expected: shortName + "-deploy",
-			},
-			{
-				base:     longName,
-				suffix:   "deploy",
-				expected: longName[:validation.DNS1123SubdomainMaxLength-16] + "-" + hash(longName) + "-deploy",
-			},
-			{
-				base:     shortName,
-				suffix:   longName,
-				expected: shortName + "-" + hash(shortName+"-"+longName),
-			},
-			{
-				base:     "",
-				suffix:   shortName,
-				expected: "-" + shortName,
-			},
-			{
-				base:     "",
-				suffix:   longName,
-				expected: "-" + hash("-"+longName),
-			},
-			{
-				base:     shortName,
-				suffix:   "",
-				expected: shortName + "-",
-			},
-			{
-				base:     longName,
-				suffix:   "",
-				expected: longName[:validation.DNS1123SubdomainMaxLength-10] + "-" + hash(longName) + "-",
-			},
-		}
+	// Any usage of longBase/longSuffix will definitely need to be truncated
+	longBase := randSeq(validation.DNS1123SubdomainMaxLength + 100)
+	longSuffix := randSeq(validation.DNS1123SubdomainMaxLength + 100)
+
+	tests := []struct {
+		name, base, suffix, expected string
+	}{
+		{
+			name:     "Short base and short suffix without truncation",
+			base:     shortBase,
+			suffix:   shortSuffix,
+			expected: shortBase + "-deploy",
+		},
+		{
+			name:     "Long base and short suffix truncates base and inserts hash",
+			base:     longBase,
+			suffix:   shortSuffix,
+			expected: longBase[:validation.DNS1123SubdomainMaxLength-hashLen-len(shortSuffix)] + "-" + hash(longBase) + "-" + shortSuffix,
+		},
+		{
+			name:     "Short base and long suffix inserts a hash instead of appending suffix",
+			base:     shortBase,
+			suffix:   longSuffix,
+			expected: shortBase + "-" + hash(shortBase+"-"+longSuffix),
+		},
+		{
+			name:     "Empty base and short suffix",
+			base:     "",
+			suffix:   shortSuffix,
+			expected: "-" + shortSuffix,
+		},
+		{
+			name:     "Empty base and long suffix appends hash of suffix",
+			base:     "",
+			suffix:   longSuffix,
+			expected: "-" + hash("-"+longSuffix),
+		},
+		{
+			name:     "Short base and no suffix",
+			base:     shortBase,
+			suffix:   "",
+			expected: shortBase + "-",
+		},
+		{
+			name:     "Long base and no suffix",
+			base:     longBase,
+			suffix:   "",
+			expected: longBase[:validation.DNS1123SubdomainMaxLength-hashLen] + "-" + hash(longBase) + "-",
+		},
+	}
 
-		for _, test := range tests {
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
 			result := GetName(test.base, test.suffix, validation.DNS1123SubdomainMaxLength)
 			if result != test.expected {
-				t.Errorf("Got unexpected result. Expected: %s Got: %s", test.expected, result)
+				t.Errorf("Got unexpected result. \nExpected: %s\nGot:      %s", test.expected, result)
 			}
```

Comment:
[latent] This could be an `assert.Equal()`

Reply:
`testify/assert` isn't a current dependency of the apis package and the code isn't too bloated since it's just string comparisons, so I'm thinking of leaving this be for now